trace_decoder/src/trace_protocol.rs

```diff
@@ -40,80 +41,124 @@ use crate::{
-    Uncompressed(TrieUncompressed),
+    Uncompressed(MptTrieUncompressed),
     /// Storage or state trie format that can be processed as is, as it
     /// corresponds to the internal format.
-    Direct(TrieDirect),
+    Direct(MptTrieDirect),
 }
 
 /// A trie pre-image where both state & storage are combined into one payload.
 #[derive(Debug, Deserialize, Serialize)]
 #[serde(rename_all = "snake_case")]
-pub struct CombinedPreImages {
+pub struct MptCombinedPreImages {
     /// Compact combined state and storage tries.
-    pub compact: TrieCompact,
+    pub compact: MptTrieCompact,
 }
 
 // TODO
 /// Bulkier format that is quicker to process.
 #[derive(Debug, Deserialize, Serialize)]
-pub struct TrieUncompressed {}
+pub struct MptTrieUncompressed {}
 
 // TODO
 #[serde_as]
 /// Compact representation of a trie (will likely be very close to <https://github.com/ledgerwatch/erigon/blob/devel/docs/programmers_guide/witness_formal_spec.md>)
 #[derive(Debug, Deserialize, Serialize)]
-pub struct TrieCompact(#[serde_as(as = "FromInto<ByteString>")] pub Vec<u8>);
+pub struct MptTrieCompact(#[serde_as(as = "FromInto<ByteString>")] pub Vec<u8>);
 
 // TODO
 /// Trie format that is in exactly the same format of our internal trie format.
 /// This is the fastest format for us to processes.
 #[derive(Debug, Deserialize, Serialize)]
-pub struct TrieDirect(pub HashedPartialTrie);
+pub struct MptTrieDirect(pub HashedPartialTrie);
 
 /// A trie pre-image where state and storage are separate.
 #[derive(Debug, Deserialize, Serialize)]
 #[serde(rename_all = "snake_case")]
-pub enum SeparateStorageTriesPreImage {
+pub enum MptSeparateStorageTriesPreImage {
     /// A single hash map that contains all node hashes from all storage tries
     /// involved in the block. We can reconstruct the individual storage tries
     /// by the storage root hash in the state entries.
-    SingleTrie(TrieUncompressed),
+    SingleTrie(MptTrieUncompressed),
 
     /// Each storage trie is sent over in a hashmap with the hashed account
     /// address as a key.
-    MultipleTries(HashMap<HashedAccountAddr, SeparateTriePreImage>),
+    MultipleTries(HashMap<HashedAccountAddr, MptSeparateTriePreImage>),
+}
+
+/// Variants on differing format for SMT.
+///
+/// Currently there is only one format, but will keep this open in case there
+/// are multiple in the future.
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum SmtBlockTraceTriePreImages {
+    /// Single trie containing all sub tries.
+    Single(SingleSmtPreImage),
+}
+
+/// Bytes for a single trie compact SMT.
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+#[serde_as]
+pub struct SingleSmtPreImage(#[serde_as(as = "FromInto<ByteString>")] Vec<u8>);
+
+/// Information that is specific to the atomic unit used by the block. This may
+/// be information for each txn or for each continuation segment.
```

Comment:
Is it worth moving this to a different pr? Mixing continuations and smt stuff makes life harder

Reply:
Sure leave it.